### ISSUE TITLE
Cashing a check should automatically create a trustline

### ISSUE THREAD

**Issue by nbougalis** [ripple/rippled#3790]
It is currently possible to send a `Check` even if the recipient doesn't have a trustline for the asset being delivered. However, attempting to cash the check will fail in the absence of a trustline.

Certain use cases would benefit from the ability to automatically create the trustline during check cashing. This is not a problem since it can only happen in response to explicit action by the recipient (similarly to how one can place an offer to purchase an asset without having a trustline established, and the system creates the trustline automatically if and when the order is filled).

The created trust line should have the limit set to 0. Optionally, the "cash" transaction could have a field that allow the recipient to manually specify a different limit.

**Comment by nbougalis** [comment]
This is in connection with https://github.com/XRPLF/XRPL-Standards/discussions/30

**Comment by mDuo13** [comment]
Agreed, and cashing a Check should allow you to accept "over" your limit amount in the same way that an Offer can.

**Comment by RichardAH** [comment]
What's the status of this? Are you still planning to introduce it as an amendment?

**Comment by nbougalis** [comment]
Yes. I did some of the prep work but the code to do this is a little more involved than I previously thought. If someone else wants to tackle it, that'd be cool, but if not, we'll probably get on in a few days.

**Comment by mDuo13** [comment]
In the meantime, another option for people who want to issue an NFT would be to offer it to the world via an OfferCreate (at some predetermined price, I suppose) and let whoever wants it claim the NFT with their own OfferCreate. Though, I suppose if you're following the "very small numbers as NFTs" proposal you'd want to make sure the rounding can't do something weird with partially-crossed offers. I don't actually know how the rounding might work if someone's offer partially crosses yours.

**Comment by develoQ** [comment]
I think this Issue can be Closed.
https://github.com/XRPLF/rippled/pull/3823
